### PR TITLE
CI: Sign test built APPX package with self-signed key

### PR DESCRIPTION
**Pull request by starnight**
To make testers test easily, sign the test APPX package with self-signed
key. Then, testers can installed the signed APPX package directly.

The Publisher in appxmanifest.xml must be the same with the
certificate's subject. And, the selfsign.pfx must match secrets'
SELFSIGN_PHRASE.

https://phabricator.endlessm.com/T33481

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -129,29 +123,27 @@ jobs:
...
-        id: sign_for_testing
-        if: github.event.inputs.sign_test_build
+      - name: Upload signed APPX for testing
+        if: steps.sign_for_testing.outcome == 'success'
+        uses: actions/upload-artifact@v2
+        with:
+          name: "endlesskeyapp.testsigned.appx"
+          path: "endlesskeyapp.testsigned.appx"
+
+      - name: Sign artifact for release
```
Comment (jprvita) [inline .github/workflows/main.yml]: We don't need to sign this for release -- it looks like the Microsoft Store accepts an unsigned package, so there is no need for this step anymore.

Comment (jprvita) [inline .github/workflows/main.yml]: We don't need to sign this for release -- it looks like the Microsoft Store accepts an unsigned package, so there is no need for this step anymore.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -163,14 +155,14 @@ jobs:
...
-            -SigningPolicySlug Endless_Apps_Test_Signing_for_Microsoft_Store `
+            -SigningPolicySlug Endless_Apps_Signing_for_Microsoft_Store `
             -ArtifactConfigurationSlug Endless_Key_APPX_Package_for_Microsoft_Store `
             -InputArtifactPath endlesskeyapp.unsigned.appx `
             -OutputArtifactPath endlesskeyapp.appx `
             -WaitForCompletion
 
-      - name: Upload signed APPX
-        if: steps.sign_for_release.outcome == 'success' || steps.sign_for_testing.outcome == 'success'
+      - name: Upload signed APPX for release
```
Comment (jprvita) [inline .github/workflows/main.yml]: Here as well, we don't need to sign anything during a release, so this step is not needed anymore.

Comment (jprvita) [inline .github/workflows/main.yml]: Here as well, we don't need to sign anything during a release, so this step is not needed anymore.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -129,29 +123,27 @@ jobs:
...
-            -OrganizationId $env:SIGNPATH_ORG_ID `
-            -ProjectSlug Endless_Apps `
-            -SigningPolicySlug Endless_Apps_Signing_for_Microsoft_Store `
-            -ArtifactConfigurationSlug Endless_Key_APPX_Package_for_Microsoft_Store `
-            -InputArtifactPath endlesskeyapp.unsigned.appx `
-            -OutputArtifactPath endlesskeyapp.appx `
-            -WaitForCompletion
+          & "${Env:ProgramFiles(x86)}\Windows Kits\10\App Certification Kit\signtool.exe" `
+            sign /fd SHA256 /a /f .buildkite_win/selfsign.pfx /p $env:SIGN_PHRASE endlesskeyapp.unsigned.appx
+          mv endlesskeyapp.unsigned.appx endlesskeyapp.testsigned.appx
```
Comment (jprvita) [inline .github/workflows/main.yml]: Let's drop the `.unsigned` part from the unsigned package (on the "Packaging" step a few lines above), since this is what we will release in the end (an unsigned package). Then maybe just call `.selfsigned` here instead of `.testsigned`?

Also, I think we should drop `app` from the name and just call it `endlesskey.appx`, but let's leave that part for a separate commit.

Comment (jprvita) [inline .github/workflows/main.yml]: Let's drop the `.unsigned` part from the unsigned package (on the "Packaging" step a few lines above), since this is what we will release in the end (an unsigned package). Then maybe just call `.selfsigned` here instead of `.testsigned`?

Also, I think we should drop `app` from the name and just call it `endlesskey.appx`, but let's leave that part for a separate commit.

**Review by jprvita — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (jprvita) [reply .github/workflows/main.yml]: Sorry, just to be clear -- we are planning to use SignPath in the future to sign the .exe for the Endless Key USB, but that should be handled in a separate ticket, as we need to define a new signing policy and artifact configuration on their platform first.

**Review by jprvita — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (starnight) [reply .github/workflows/main.yml]: ok

File: .github/workflows/main.yml (hunk 1 above)
Comment (starnight) [reply .github/workflows/main.yml]: ok

Comment (starnight) [reply .github/workflows/main.yml]: ok

File: .github/workflows/main.yml (hunk 2 above)
Comment (starnight) [reply .github/workflows/main.yml]: ok

File: .github/workflows/main.yml (hunk 3 above)
Comment (starnight) [reply .github/workflows/main.yml]: ok

File: .github/workflows/main.yml (hunk 2 above)
Comment (starnight) [reply .github/workflows/main.yml]: ok

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -121,63 +115,35 @@ jobs:
       - name: Packaging
         run: |
           cp .buildkite_win/appxmanifest.xml kolibri-windows/
-          & "${Env:ProgramFiles(x86)}\Windows Kits\10\App Certification Kit\makeappx.exe" pack /d kolibri-windows /p endlesskeyapp.unsigned.appx
-
-      - name: Upload unsigned APPX
```
Comment (jprvita) [inline .github/workflows/main.yml:126]: Let's keep this stage uploading the unsigned package for all builds, so we can easily get it for checking if it pass validation with the store.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -121,63 +115,35 @@ jobs:
...
+          mv endlesskey.appx endlesskey.selfsigned.appx
+
+      - name: Upload self-signed package for testing
+        if: steps.sign_for_testing.outcome == 'success'
+        uses: actions/upload-artifact@v2
+        with:
+          name: "endlesskey.selfsigned.appx"
+          path: "endlesskey.selfsigned.appx"
+
+      - name: Upload unsigned package for release
```
Comment (jprvita) [inline .github/workflows/main.yml]: And since we will upload the unsigned package on every build, before signing, we don't need this stage on releases (only the next "Release" stage to create the release on GitHub).

**Review by jprvita — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 4 above)
Comment (starnight) [reply .github/workflows/main.yml:126]: ok

File: .github/workflows/main.yml (hunk 5 above)
Comment (starnight) [reply .github/workflows/main.yml]: ok

**Review by jprvita — APPROVED** [review]
